### PR TITLE
Use cobra for the kmd command to allow for documentation automation.

### PR DESCRIPTION
**Pull request by winder**

### CODE REVIEW

File: cmd/kmd/main.go [hunk 1]
```diff
@@ -36,30 +38,43 @@ const (
...
+}
 
+var kmdCmd = &cobra.Command{
+	Use:   "kmd",
+	Short: "Key Management Daemon (KMD)",
+	Long:  `The Key Management Daemon (KMD) is a low level wallet and participation
+key management tool. It works in conjunction with algod and goal to keep
+secrets safe. An optional timeout flag will automatically terminate KMD
+after a number of seconds has elapsed, allowing a simple way to ensure KMD
+will be shutdown in a timely manner. This is a blocking command that.`,
```
Comment (justicz) [inline cmd/kmd/main.go]: typo: "This is a blocking command that."
nit: kmd doesn't manage participation keys
nit: can you please replace all instances of KMD with kmd?

File: cmd/kmd/main.go [hunk 2]
```diff
@@ -36,30 +38,43 @@ const (
...
+
+func runKmd(dataDir string, timeoutSecs uint64) {
 	// Use logging package instead of stdin/stdout
 	log := logging.NewLogger()
 	log.SetLevel(logging.Info)
 
-	// Validate flags
-	if *dataDir == "" {
-		log.Errorf("dataDir (-d) is a required argument")
-		os.Exit(codes.ExitCodeKMDInvalidArgs)
```
Comment (justicz) [inline cmd/kmd/main.go]: please delete `ExitCodeKMDInvalidArgs` if it's not used anywhere

**Comment by justicz** [comment]
coupla nits, overall looks good!

File: cmd/kmd/main.go [hunk 3]
```diff
@@ -36,30 +38,43 @@ const (
...
+	},
+}
+
+func runKmd(dataDir string, timeoutSecs uint64) {
 	// Use logging package instead of stdin/stdout
 	log := logging.NewLogger()
 	log.SetLevel(logging.Info)
 
-	// Validate flags
-	if *dataDir == "" {
```
Comment (pzbitskiy) [inline cmd/kmd/main.go]: Add kmdCmd.MarkFlagRequired("data-dir") to preserve current behavior?

**Pull request merged**